contracts/src/core/libraries/v3/config-registry/ConfigRegistryLib.sol

```diff
@@ -51,34 +64,12 @@ library ConfigRegistryLib {
         registerKey(S.layout().bytesConfig.nameSet, name);
     }
 
-    function deregisterKeyBytes32(string memory name) internal {
-        deregisterKey(S.layout().bytes32Config.nameSet, name);
-    }
-
-    function deregisterKeyBytes(string memory name) internal {
-        deregisterKey(S.layout().bytesConfig.nameSet, name);
-    }
-
     function registerKey(T.NameSet storage nameSet, string memory name) internal {
         bytes32 key = getKey(name);
-        require(bytes(nameSet.names[key]).length == 0, "Key already exists");
-        require(bytes(name).length > 0, "Name cannot be empty");
-        nameSet.names[key] = name;
-        nameSet.nameList.push(name);
-    }
-
-    function deregisterKey(T.NameSet storage nameSet, string memory name) internal {
-        bytes32 key = getKey(name);
-        require(bytes(nameSet.names[key]).length > 0, "Key does not exist");
-        delete nameSet.names[key];
-        // Here we utilize a simple swap and pop to remove the name from the list.
-        // There is no guarantee of preservation of ordering.
-        for (uint256 i; i < nameSet.nameList.length; i++) {
-            if (getKey(nameSet.nameList[i]) == key) {
-                nameSet.nameList[i] = nameSet.nameList[nameSet.nameList.length - 1];
-                nameSet.nameList.pop();
-                break;
-            }
+        if (bytes(nameSet.names[key]).length == 0) {
+            require(bytes(name).length > 0, "Name cannot be empty");
```

Comment:
shouldnt this check be done as the very first thing even before calling `getKey(name)`?

Reply:
Yes, that makes sense logically. However, I don't think it's ideal to be increasing the gas cost for any view getters to this call, and I don't think there's any real security concerns as long as this check is enforced upon name registration. Can you think of a security issue with doing this check here instead of the more load bearing function?